lib/runner/index.js

```diff
@@ -3,35 +3,34 @@ var _ = require('lodash'),
     Run = require('./run'),
     extractRunnableItems = require('./extract-runnable-items').extractRunnableItems,
 
-    DEFAULT_GLOBAL_TIMEOUT = 3 * 60 * 1000, // 3 minutes
     Runner,
 
-    /**
-     * Converts any non finite values and 0 or negative values to Infinity.
-     *
-     * @param {Number} ms
-     *
-     * @returns {Number}
-     */
-    sanitizeTimeout = function (ms) {
-        return (_.isFinite(ms) && ms > 0) ? ms : Infinity;
+    defaultTimeouts = {
+        global: 3 * 60 * 1000, // 3 minutes
+        request: Infinity,
+        script: Infinity
     },
 
     /**
-     * Sanitizes `timeout` `options` if any on the options object.
+     * Sanitizes run options. Only handles timeout for now.
      *
-     * @param {Object} [timeoutOptions]
+     * @param {Object} options
      *
-     * @returns {Object|undefined}
+     * @returns {Object}
      */
-    sanitizeTimeouts = function (timeoutOptions) {
-        if (!timeoutOptions) { return; }
+    sanitizeRunOptions = function (options) {
+        !options.timeout && (options.timeout = {});
 
-        _.forEach(timeoutOptions, function (value, key) {
-            timeoutOptions[key] = sanitizeTimeout(value);
+        // start timeout sanitization
+        _.mergeWith(options.timeout, defaultTimeouts, function (userTimeout, defaultTimeout) {
+            // non numbers, Infinity and missing values are set to default
+            if (!_.isFinite(userTimeout)) { return defaultTimeout; }
+
+            // 0 and negative numbers are set to Infinity, which only leaves positive numbers
+            return userTimeout > 0 ? userTimeout : Infinity;
```

Comment:
@shamasis @kamalaknn Just wanted to bring attention to this
Ideally, this should not work for `vm`. 
It working coz Infinity is being converted to `null` during dispatching.